sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/signaling/TrouterUtils.java

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.chat.signaling;
+
+import com.azure.android.communication.chat.signaling.properties.ChatEventId;
+import com.azure.android.core.logging.ClientLogger;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+class TrouterUtils {
+
+    private static final ClientLogger CLIENT_LOGGER = new ClientLogger(TrouterUtils.class);
+    /**
+     * Mapping chat event id to trouter event id code
+     */
+    static final Map<ChatEventId, Integer> EVENT_IDS_MAPPING = new HashMap<ChatEventId, Integer>() {{
+            put(ChatEventId.chatMessageReceived, 200);
+            put(ChatEventId.typingIndicatorReceived, 245);
+            put(ChatEventId.readReceiptReceived, 246);
+            put(ChatEventId.chatMessageEdited, 247);
+            put(ChatEventId.chatMessageDeleted, 248);
+            put(ChatEventId.chatThreadCreated, 257);
+            put(ChatEventId.chatThreadPropertiesUpdated, 258);
+            put(ChatEventId.chatThreadDeleted, 259);
+            put(ChatEventId.participantsAdded, 260);
+            put(ChatEventId.participantsRemoved, 261);
+        }};
+
+    public static JSONObject toMessageHandler(String chatEventId, String responseBody) {
+        int eventId = 0;
+        JSONObject genericPayload = null;
+        try {
+            eventId = EVENT_IDS_MAPPING.get(ChatEventId.valueOf(chatEventId));
+            genericPayload = new JSONObject(responseBody);
+        } catch (Exception e) {
+            CLIENT_LOGGER.error(e.getMessage());
+        }
+
+        try {
+            if (genericPayload == null || genericPayload.getInt("_eventId") != eventId) {
+                return null;
+            }
+        } catch (JSONException e) {
+            CLIENT_LOGGER.error(e.getMessage());
+        }
+
+        return toEventPayload(chatEventId, genericPayload);
+    }
+
+    public static JSONObject toEventPayload(String eventId, JSONObject payload) {
+        if (eventId.equals(ChatEventId.chatMessageReceived.name())) {
```

Comment:
Consider refactor the block into a small private function. Easier to read that way.